blender/LilySurfaceScraper/Scrapers/AbstractScraper.py

```diff
@@ -102,61 +112,61 @@ def getTextureDirectory(self, material_name):
             texture_dir = texture_dir[2:]
         if not os.path.isabs(texture_dir):
             texture_dir = os.path.realpath(os.path.join(self.texture_root, texture_dir))
-        name_path = material_name.replace('/',os.path.sep)
+        name_path = material_name.replace('/', os.path.sep)
         dirpath = os.path.join(texture_dir, name_path)
         if not os.path.isdir(dirpath):
             os.makedirs(dirpath)
         return dirpath
 
-    def fetchImage(self, url, material_name, map_name, force_ext=False):
-        """Utility helper for download textures"""
-        root = self.getTextureDirectory(material_name)
-        if not force_ext:
-            ext = os.path.splitext(url)[1]
-            map_name = map_name + ext
-        path = os.path.join(root, map_name)
-        if os.path.isfile(path):
-            print("Using cached {}.".format(url))
-        else:
-            print("Downloading {}...".format(url))
-            headers = {"User-Agent":"Mozilla/5.0"}  # fake user agent
+    def downloadFunc(self, url, path):
+        def func():
+            headers = {"User-Agent": "Mozilla/5.0"}  # fake user agent
             r = requests.get(url, stream=True, headers=headers)
             if r.status_code == 200:
                 with open(path, 'wb') as f:
                     r.raw.decode_content = True
                     shutil.copyfileobj(r.raw, f)
             else:
                 self.error = "URL not found: {}".format(url)
-                return None
-        return path
+                return -1
+        return func
+
+    def fetchImage(self, url, material_name, map_name, force_ext=False):
+        """Utility helper for download textures"""
+        root = self.getTextureDirectory(material_name)
+        if not force_ext:
+            ext = os.path.splitext(url)[1]
+            map_name = map_name + ext
+        path = os.path.join(root, map_name)
+        return self.saveFile(path, self.downloadFunc(url, path))
 
     def fetchFile(self, url, material_name, filename):
         root = self.getTextureDirectory(material_name)
         path = os.path.join(root, filename)
-        if os.path.isfile(path):
-            return path
-        data = self._fetch(url)
-        with open(path, "wb") as f:
-            f.write(data.content)
-        data.close()
-        return path
+
+        def saveFile():
+            data = self._fetch(url)
+            with open(path, "wb") as f:
+                f.write(data.content)
+            data.close()
+
+        return self.saveFile(path, saveFile)
 
     def fetchZip(self, url, material_name, zip_name):
         """Utility helper for download textures"""
         root = self.getTextureDirectory(material_name)
         path = os.path.join(root, zip_name)
-        if os.path.isfile(path):
-            print("Using cached {}.".format(url))
+        return self.saveFile(path, self.downloadFunc(url, path))
+
+    def saveFile(self, path, getData):
```

Comment:
Maybe `ensureFile` would be a better name. Also prefix with an underscore all methods that are meant to be private, i.e. not used in subclasses.